packages/FormElement/src/components/Label/Label.js

```diff
@@ -4,56 +4,68 @@ import PropTypes from "prop-types";
 import useI18n from "@paprika/l10n/lib/useI18n";
 
 import labelStyles, { ruleStyles } from "./Label.styles";
+import Help from "../Help/Help";
+import { FormElementContext } from "../../FormElement";
 
 const propTypes = {
+  children: PropTypes.node.isRequired,
   /** If "optional" text should be displayed beside the label  */
-  hasOptionalLabel: PropTypes.bool.isRequired,
+  hasOptionalLabel: PropTypes.bool,
   /** If "require" text should be displayed beside the label */
-  hasRequiredLabel: PropTypes.bool.isRequired,
+  hasRequiredLabel: PropTypes.bool,
   /** Help indicator */
   help: PropTypes.node,
   /** id for the element */
   id: PropTypes.string,
-  /** Should label and children be inline or not */
-  isInline: PropTypes.bool.isRequired,
   /** Should label be hidden */
-  isVisuallyHidden: PropTypes.bool.isRequired,
-  /** Label text of the input field */
-  label: PropTypes.node.isRequired,
-  /** Set if FormElement contains multiple children to render a legend element instead of label */
-  hasFieldSet: PropTypes.bool.isRequired,
+  isVisuallyHidden: PropTypes.bool,
 
   onClick: PropTypes.func,
 };
 
 const defaultProps = {
+  hasOptionalLabel: false,
+  hasRequiredLabel: false,
   help: null,
   id: null,
+  isVisuallyHidden: false,
   onClick: () => {},
 };
 
-const Label = React.forwardRef((props, ref) => {
-  const { hasOptionalLabel, hasRequiredLabel, help, id, hasFieldSet, label, ...moreProps } = props;
+const Label = props => {
+  const { hasFieldSet, refLabel, labelId } = React.useContext(FormElementContext);
+  const { hasOptionalLabel, hasRequiredLabel, help, id, children, ...moreProps } = props;
 
   const I18n = useI18n();
 
-  const labelProps = hasFieldSet ? { as: "legend" } : { as: "label", htmlFor: id, ref };
+  const labelProps = hasFieldSet ? { as: "legend" } : { as: "label" };
+
+  const a11yProps = { htmlFor: labelId, ref: refLabel };
+
+  const renderQuestionRequirement = () => {
+    if (hasRequiredLabel) {
+      return I18n.t("formElement.required");
+    }
+    if (hasOptionalLabel) {
+      return I18n.t("formElement.optional");
+    }
+  };
+
+  const renderHelp = () => {
+    if (help) {
+      return <Help>{help}</Help>;
+    }
+    return null;
+  };
 
   return (
-    <div data-pka-anchor="form-element.label" css={labelStyles} {...labelProps} {...moreProps}>
-      {label}
-      {hasOptionalLabel || hasRequiredLabel ? (
-        <span css={ruleStyles}>
-          {" "}
-          {hasRequiredLabel ? I18n.t("formElement.required") : null}
-          {hasOptionalLabel ? I18n.t("formElement.optional") : null}
-        </span>
-      ) : null}
-
-      {help}
+    <div data-pka-anchor="form-element.label" css={labelStyles} {...labelProps} {...moreProps} {...a11yProps}>
+      {children}
+      {hasOptionalLabel || hasRequiredLabel ? <span css={ruleStyles}> {renderQuestionRequirement()}</span> : null}
+      {renderHelp()}
```

Comment:
This is minor, and can be addressed later, but since the `<label>` `<legend>` wraps the `<Help>` trigger, the "Information" label for that trigger gets appended to label that the screen reader announces, which can be a bit confusing.  

Wrapping the whole label in a `<div>` and then just the `children` and "required/optional" bit in the `<label>` needs to be thought through though, because then were would the `moreProps` go? Probably have to go on the `div`, but then you have no way of putting more props on the `label` without a `propsCollector`.  That _might_ be okay, but not sure.  So yeah, okay for now.

Reply:
👌